### ISSUE TITLE
Pressing skip while moving teleports the catcher (in that direction)

### ISSUE THREAD

**Issue by KevKjef** [ppy/osu#9526]
**Describe the bug:** 
If you move the catcher in a direction and then press skip to start the song, the catcher teleports in that direction

**Screenshots or videos showing encountered issue:** 
https://twitter.com/osuChatie/status/1281647054612074497
This happens on all maps with a skip function

**osu!lazer version:** 
2020.710.0

**Comment by bdach** [comment]
To clarify, this can be reproducing by skipping to start of gameplay while holding any of the direction keys.

This is an unintended consequence of how gameplay clocks work. In terms of the gameplay clock, skipping the intro is tantamount to seeking forward to a frame far in the future. The catcher is using the gameplay clock's elapsed time to compute its frame-stable velocity of movement:

https://github.com/ppy/osu/blob/4198f4d96e88f5d75def4cd21acad59c9e1193e7/osu.Game.Rulesets.Catch/UI/Catcher.cs#L408

which in this case leads to the effect mentioned, as the time delta is large enough so that the catcher ends up on either side of the screen.

To alleviate this, the catcher would probably have to use a frame-stable clock which isn't inherently tied to the track position.

**Comment by peppy** [comment]
I'm not sure how incorrect this is, does it feel bad? Would it feel better if there was a trail or soemthing?

I was going to suggest a temporary/permanent solution of ignoring large elapsed time values but I think the fact we are in a `FrameStable` context makes that not possible.

**Comment by KevKjef** [comment]
Well, it just feels weird to 'suddenly' teleport to the other side of the screen. It doesn't affect gameplay too much, but if I move the catcher left-right a lot on say the right side of the screen (since I know the map starts there) and then press skip, which teleports me to the left, there is a reasonable chance I would miss the first note (and yes this can be prevented by not moving the catcher, but it is something I (and many others) like to do)
I think having a trail would make it a bit better, but it wouldn't completely fix the issue. (+it is kind of hard to say what kind of trail would work)

**Comment by bdach** [comment]
For what it's worth other gameplay elements in other places will also sort of be affected by this kind of thing (for example, if an osu! skin has a spinning cursor, its rotation will jerk forward when intro is skipped). Not anywhere near as disruptive as this, but still.

**Comment by peppy** [comment]
I think an ultimate solution for this may be to make skip tween forward, rather than instantaneously jump.

**Comment by Flutterish** [comment]
What if when a skip is initiated, all user inputs zero out, then anything that is supposed to use these inputs is updated ( before changing the clock )? This will not fix the skin spin jump, but time interpolated values that depend on user input will be fixed.

**Comment by peppy** [comment]
I disagree with that approach. Skipping is skipping gameplay and as such all elements should also skip accordingly (how it is currently happening).

**Comment by Flutterish** [comment]
Not everything is a gameplay element though. As bdach pointed out - skins. I think at least they should have an absolute-time based animations.

As per gameplay elements jumping on a skip, it might be halpful to add an `OnSkip` method to `DrawableHitObject`s, and `Playfield`s instead.

**Comment by peppy** [comment]
Everything in gameplay container is gameplay. The path forward is approaching this with my above solution and iterating from there. Further discussion is not required.
